testing/rspec/spec/unit/linters/unique_scenario_names_linter_spec.rb

```diff
@@ -0,0 +1,257 @@
+RSpec.describe CukeLinter::UniqueScenarioNamesLinter do
+  let(:linter) { described_class.new }
+
+  before do
+    described_class.instance_variable_set(:@scenario_names, {})
+  end
+
+  context 'when there are duplicate scenario names within a feature file' do
+    let(:feature_file) do
+      """
+      Feature: Sample Feature
+
+        Scenario: Duplicate Scenario Name
+          Given something
+
+        Scenario: Duplicate Scenario Name
+          Given something else
+      """
+    end
+
+    let(:model) do
+      generate_feature_model(source_text: feature_file, parent_file_path: 'path_to_file').tests
+    end
+
+    it 'detects duplicate scenario names' do
+      results = model.map { |scenario| linter.lint(scenario) }.compact
+      expect(results).to match_array([
+                                       { linter:   'UniqueScenarioNamesLinter',
+                                         problem:  'Scenario names are not unique',
+                                         location: 'path_to_file:7' }
+                                     ])
+    end
+  end
+
+  context 'when a scenario outline generates duplicate names' do
+    let(:feature_file) do
+      """
+      Feature: Sample Feature with Scenario Outline
+
+        Scenario Outline: Repeated Scenario Name Doing <input>
+
+        Examples:
+          | input          |
+          | Something      |
+          | Something      |
+      """
+    end
+
+    let(:model) do
+      generate_feature_model(source_text: feature_file, parent_file_path: 'path_to_file').tests
+    end
+
+    it 'reports duplicate scenario names generated by the scenario outline' do
+      results = model.map { |scenario| linter.lint(scenario) }.compact
+      expect(results).to include(
+        {
+          linter:   'UniqueScenarioNamesLinter',
+          problem:  'Template creates scenario names that are not unique',
+          location: 'path_to_file:4'
+        }
+      )
+    end
+  end
+
+  context 'when a scenario outline generates a name identical to a regular scenario' do
+    let(:feature_file) do
+      """
+      Feature: Feature with Mixed Scenarios
+
+        Scenario: Duplicate Scenario Name
+          Given something
+
+        Scenario Outline: Duplicate Scenario Name
+          Given something
+
+        Examples:
+          | input          |
+          | Something      |
+      """
+    end
+
+    let(:model) do
+      generate_feature_model(source_text: feature_file, parent_file_path: 'path_to_file').tests
+    end
+
+    it 'reports duplicate names generated by the scenario outline and the regular scenario' do
```

Comment:
There is duplicity in context and it. Here and other places

```
context 'when a scenario outline generates a name identical to a regular scenario' do       
  it 'reports duplicate names generated by the scenario outline and the regular scenario' do
```

Reply:
I changed the descriptive part towards the style of existing tests and got this output of the spec test

```
CukeLinter::UniqueScenarioNamesLinter
  has a name
  it should behave like a linter at the unit level
    is named
    can lint
    lints a model
  linting
    with the same scenario names across different feature files
      returns no problem
    with duplicate scenario names within a feature file
      returns a detected problem and its location
    with a duplicate scenario names generated by scenario outline
      returns a detected problem and its location
    with an outline generated scenario name is identical to a regular scenario
      returns a detected problem and its location
    with identical scenario names generated by different outlines
      returns a detected problem and its location
    with duplicate names created by scenario outline with no placeholders
      returns a detected problem and its location
    with duplicate scenario names within a Rule
      returns a detected problem and its location
    with duplicate scenario names across multiple Rules within a feature
      returns a detected problem and its location
```

What do you think?

Reply:
```
CukeLinter::SingleTestBackgroundLinter
  has a name
  it should behave like a linter at the unit level
    is named
    can lint
    lints a model
  linting
    with a background that affects only one test
      and that test is a scenario
        records a problem
        it should behave like a linter linting a bad model
          returns a detected problem
          includes the problem and its location in its result
          correctly records the location of the problem
      and that test is a outline
        records a problem
        it should behave like a linter linting a bad model
          returns a detected problem
          includes the problem and its location in its result
          correctly records the location of the problem
    with a background that affects multiple tests
      it should behave like a linter linting a good model
        returns no problem
    with a background that affects no tests
      it should behave like a linter linting a good model
        returns no problem
    with a non-background model
      it should behave like a linter linting a good model
        returns no problem
```

```
CukeLinter::StepWithEndPeriodLinter
  has a name
  it should behave like a linter at the unit level
    is named
    can lint
    lints a model
  linting
    with a step that ends with a period
      records a problem
      it should behave like a linter linting a bad model
        returns a detected problem
        includes the problem and its location in its result
        correctly records the location of the problem
    with a step that does not end with a period
      it should behave like a linter linting a good model
        returns no problem
    with a non-step model
      it should behave like a linter linting a good model
        returns no problem
```

```
CukeLinter::TestWithNoNameLinter
  has a name
  it should behave like a linter at the unit level
    is named
    can lint
    lints a model
  linting
    with a scenario that has no name
      because its name is empty
        records a problem
        it should behave like a linter linting a bad model
          returns a detected problem
          includes the problem and its location in its result
          correctly records the location of the problem
      because its name is nil
        records a problem
        it should behave like a linter linting a bad model
          returns a detected problem
          includes the problem and its location in its result
          correctly records the location of the problem
    with a scenario that does have a name
      it should behave like a linter linting a good model
        returns no problem
    with a outline that has no name
      because its name is empty
        records a problem
        it should behave like a linter linting a bad model
          returns a detected problem
          includes the problem and its location in its result
          correctly records the location of the problem
      because its name is nil
        records a problem
        it should behave like a linter linting a bad model
          returns a detected problem
          includes the problem and its location in its result
          correctly records the location of the problem
    with a outline that does have a name
      it should behave like a linter linting a good model
        returns no problem
    a non-test model
      it should behave like a linter linting a good model
        returns no problem
```

Reply:
Looks like you slightly changes the style. I propose to unify style

Reply:
```
CukeLinter::UniqueScenarioNamesLinter
  has a name
  it should behave like a linter at the unit level
    is named
    can lint
    lints a model
  linting
    with scenarios that have unique names within the same feature file
      it should behave like a linter linting a good model
        returns no problem
    with duplicate scenario names across different feature files
      it should behave like a linter linting a good model
        returns no problem
    with a scenario outline generating unique names
      it should behave like a linter linting a good model
        returns no problem
    with a scenario that has a duplicate name within the same feature file
      returns a detected problem
      includes the problem and its location in its result
    with a scenario outline generating duplicate names
      when duplicates are generated by the same outline
        returns a detected problem
        includes the problem and its location in its result
      when duplicates are generated between an outline and a regular scenario
        returns a detected problem
        includes the problem and its location in its result
      when duplicates are generated by different outlines
        returns a detected problem
        includes the problem and its location in its result
      when the outline has no placeholders in the name
        returns a detected problem
        includes the problem and its location in its result
```